server/protocols/webadmin/webadmin-cassandra/src/main/java/org/apache/james/webadmin/services/TasksCleanupService.java

```diff
@@ -0,0 +1,113 @@
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.services;
+
+import java.time.Instant;
+import java.util.concurrent.atomic.AtomicLong;
+
+import javax.inject.Inject;
+
+import org.apache.james.eventsourcing.eventstore.EventStore;
+import org.apache.james.task.Task;
+import org.apache.james.task.eventsourcing.TaskAggregateId;
+import org.apache.james.task.eventsourcing.TaskExecutionDetailsProjection;
+import org.apache.james.util.ReactorUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class TasksCleanupService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TasksCleanupService.class);
+
+    public static class Context {
+
+        public static class Snapshot {
+            private final long removedTasksCount;
+            private final long processedTaskCount;
+
+            public Snapshot(long removedTasksCount, long processedTaskCount) {
+                this.removedTasksCount = removedTasksCount;
+                this.processedTaskCount = processedTaskCount;
+            }
+
+            public long getRemovedTasksCount() {
+                return removedTasksCount;
+            }
+
+            public long getProcessedTaskCount() {
+                return processedTaskCount;
+            }
+        }
+
+        private final AtomicLong removedTasksCount;
+        private final AtomicLong processedTaskCount;
+
+        public Context() {
+            removedTasksCount = new AtomicLong();
+            processedTaskCount = new AtomicLong();
+        }
+
+        void incrementRemovedTasksCount() {
+            removedTasksCount.incrementAndGet();
+        }
+
+        void incrementProcessedTaskCount() {
+            processedTaskCount.incrementAndGet();
+        }
+
+        void incrementRemovedTasksCount(int count) {
+            removedTasksCount.set(removedTasksCount.get() + count);
+        }
+
+        public Snapshot snapshot() {
+            return new Snapshot(removedTasksCount.get(), processedTaskCount.get());
+        }
+    }
+
+    private final EventStore eventStore;
+    private final TaskExecutionDetailsProjection taskExecutionDetailsProjection;
+
+    @Inject
+    public TasksCleanupService(EventStore eventStore, TaskExecutionDetailsProjection taskExecutionDetailsProjection) {
+        this.eventStore = eventStore;
+        this.taskExecutionDetailsProjection = taskExecutionDetailsProjection;
+    }
+
+    public Mono<Task.Result> removeBeforeDate(Instant beforeDate, Context context) {
+        return Flux.from(taskExecutionDetailsProjection.listDetailsByBeforeDate(beforeDate))
+            .doOnNext(oldTaskDetail -> context.incrementProcessedTaskCount())
+            .flatMap(taskExecutionDetails -> Mono.from(eventStore.remove(new TaskAggregateId(taskExecutionDetails.taskId())))
```

Comment:
```suggestion
        return Flux.from(taskExecutionDetailsProjection.listDetailsByBeforeDate(beforeDate))
            .doOnNext(oldTaskDetail -> context.incrementProcessedTaskCount())
            .flatMap(taskExecutionDetails -> Mono.from(eventStore.remove(new TaskAggregateId(taskExecutionDetails.taskId())))
            
```

How about to move this "deletion logic" into EventSourcingTaskManager ?